packages/rocketchat-api/server/v1/channels.js

```diff
@@ -480,6 +582,40 @@ RocketChat.API.v1.addRoute('channels.setPurpose', { authRequired: true }, {
 	}
 });
 
+RocketChat.API.v1.addRoute('channels.setReadOnly', { authRequired: true }, {
+	post: function() {
+		if (typeof this.bodyParams.readOnly === 'undefined') {
+			return RocketChat.API.v1.failure('The bodyParam "readOnly" is required');
+		}
+
+		const findResult = findChannelById(this.bodyParams.roomId);
+
+		if (findResult.statusCode) {
+			return findResult;
+		}
+
+		if (findResult.archived) {
```

Comment:
Add this check to the `findChannelById` via passing in an `options` default to true.